lib/SIL/GraphFunctionDeviceInfo.cpp

```diff
@@ -165,9 +217,8 @@ DeviceType GraphFunctionDeviceInfo::chooseDevice(llvm::StringRef opType) const {
       opType == "WriteScalarSummary")
     return DeviceType::CPU;
 
-  // Place this inst on the device given by this deviceInfo.
-  // FIXME: Use the op kernel device availability info to select a device for
-  // `opType` -- if that op has no available kernel on `primaryDeviceType`, a
-  // different device should be returned.
-  return primaryDeviceType;
+  if (primaryDeviceType == DeviceType::GPU && CanRunOnDevice(opType, attributes, "GPU") {
```

Comment:
do you think we are ready to remove all of the hacked up code above in this func, at least in the case of `primaryDeviceType == DeviceType::GPU`? 

As we may still need them for the TPU case, maybe wrapped them under an if test over `primaryDeviceType == DeviceType::TPU`.

overall the code here may need further extension for TPU. Pls feel free to leave a TODO for now.